common/include/pcl/common/impl/io.hpp

```diff
@@ -130,16 +132,28 @@ copyPointCloud (const pcl::PointCloud<PointInT> &cloud_in,
   cloud_out.sensor_origin_ = cloud_in.sensor_origin_;
   cloud_out.points.resize (cloud_in.size ());
 
-  if (cloud_in.points.empty ())
+  if (cloud_in.points.empty ()) {
     return;
+  }
 
-  if (isSamePointType<PointInT, PointOutT> ())
-    // Copy the whole memory block
-    memcpy (&cloud_out[0], &cloud_in[0], cloud_in.size () * sizeof (PointInT));
+#if (__cplusplus == 201703L)
```

Comment:
Please use `PCL_MACROS` for a `PCL_CONSTEXPR` like solution.

Moreover, why did you stick with memcpy in non C++17 branch?

Reply:
This relates to the [issue I mentioned](https://github.com/PointCloudLibrary/pcl/issues/3368#issuecomment-664730021) where compilation failed due to trying to use `std::copy` on two different `Point` objects, despite the `isSamePoint` test

Reply:
Ofc. Silly of me. Please let this issue still remain unresolved till I get to think over it a bit